source/Octopus.Tentacle/Services/Capabilities/CapabilitiesServiceV2.cs

```diff
@@ -15,7 +15,15 @@ public class CapabilitiesServiceV2 : IAsyncCapabilitiesServiceV2
         public async Task<CapabilitiesResponseV2> GetCapabilitiesAsync(CancellationToken cancellationToken)
         {
             await Task.CompletedTask;
-            return new CapabilitiesResponseV2(new List<string>() {nameof(IScriptService), nameof(IFileTransferService), nameof(IScriptServiceV2), nameof(IScriptServiceV3Alpha)});
+
+            //the kubernetes agent only supports the kubernetes script services
+            if (PlatformDetection.Kubernetes.IsRunningAsKubernetesAgent)
+            {
+                return new CapabilitiesResponseV2(new List<string> { nameof(IFileTransferService), nameof(IKubernetesScriptServiceV1Alpha) });
+            }
+
+            //non-kubernetes agent tentacles only support the standard script services
+            return new CapabilitiesResponseV2(new List<string> { nameof(IScriptService), nameof(IFileTransferService), nameof(IScriptServiceV2) });
```

Comment:
The Capabilities response now returns difference script services based on if running as a Kubernetes agent or not